content/lessons/10_smart_pointers/ref_count.rs

```diff
@@ -0,0 +1,31 @@
+use std::rc::Rc;
+
+struct LoudInt(i32);
+
+impl Drop for LoudInt {
+    fn drop(&mut self) {
+        println!("[{}] Farewell!", self.0);
+    }
+}
+
+fn main() {
+    {
+        let outer_ref;
+
+        {
+            let inner_ref = Rc::new(LoudInt(5));
+
+            // strong_count represents the number of owning references pointing
+            // to data
+            assert_eq!(Rc::strong_count(&inner_ref), 1);
+
+            outer_ref = Rc::clone(&inner_ref);
+
+            assert_eq!(Rc::strong_count(&inner_ref), Rc::strong_count(&outer_ref));
+            assert_eq!(Rc::strong_count(&inner_ref), 2);
+        }
+
+        println!("The {} still lives!", outer_ref.0);
+        assert_eq!(Rc::strong_count(&outer_ref), 1);
```

Comment:
z tym liczeniem to pamiętam że coś dziwnego było

Reply:
https://www.reddit.com/r/rust/comments/td3utn/why_does_weakweak_count_return_0_when_strong_is_0/